### DISCUSSION TITLE
Question: Using code-server with nginx and base path appended

### DISCUSSION THREAD

**Discussion by Lesmiscore** [coder/code-server#1739] [Q&A] [answered]
I expose (old) code-server via nginx, with the base path set. (e.g. expose code-server at `/vs-code/`)      
(SSL is configured at nginx, so I don't need SSL on `code-server`.)    
Is it possible to use `code-server` with the base path in the current version? If it's possible, how should I configure `code-server`?

**Comment by nhooyr** [comment]
cc @code-asher

**Comment by nathanrpage97** [comment]
@nao20010128nao if code server doesn't have the ability to set base path, you can always setup nginx to [rewrite requests](https://gist.github.com/soheilhy/8b94347ff8336d971ad0#step-7----rewriting-requests).

**Comment by dtsmith2001** [comment]
See the Deep Ocean tutorial I put in #1741

**Comment by code-asher** [comment, marked as answer]
Yup, the critical part is the trailing slash on `proxy_pass`.
```
    location /vs-code/ {
      proxy_pass http://localhost:8080/;
```
That's what makes nginx rewrite the URL without the base path.

**Comment by nhooyr** [comment]
Makes sense, all our links are relative otherwise so that's why `--base-path` was removed.

**Comment by github0null** [comment]
We need a proxy_redirect to rewrite the 302 request returned by code-server to avoid routing to the wrong path.

Because code-server will return a 302 response with `Location: ./../login` when open `https://your.domain.com/code-server/`

So the complete configuration should be like this:

```
location /code-server/ {
            proxy_pass http://localhost:8080/;
            proxy_set_header Host $host;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection upgrade;
            proxy_set_header Accept-Encoding gzip;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            # redirect rewrite
            proxy_redirect ~^\./\.\./(.*)$ /code-server/$1;
            proxy_redirect / /code-server/;
}
```

Then open `https://your.domain.com/code-server/` to access

**Reply by code-asher** [reply]
Hmm strange, when you use a trailing slash in `proxy_pass`, NGINX is supposed to rewrite the base path out of the request, so code-server should just see `/` and redirect to `./login`, which makes `https://your.domain.com/code-server/` go to `https://your.domain.com/code-server/login`.

Maybe we have a bug in generating that relative path, although I am not sure what it could be.  Or maybe NGINX is sending the wrong request path, but that seems less likely.
